tests/hover.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { expect, test } from "@playwright/test";
+
+import { AllPages } from "../pages/allpages";
+
+test.describe("hover", () => {
+  let pages: AllPages;
+
+  test.beforeEach(async ({ page }) => {
+    pages = new AllPages(page);
+    await pages.hoverPage.visit();
+  });
+
+  test("template for you", async ({ page }) => {
+    //pises kod sem :)
+  });
+
+  test("on the hover page are 3 animal cards", async ({ page }) => {
+    await expect(pages.hoverPage.animalCard).toHaveCount(3);
+  });
+
+  test("when hover on cat card it shows right title", async ({ page }) => {

```

Comment:
Zkusil jsem zakomentoval Tvoje kroky a ono to porad da `PASS` - jakto? Dokazes to opravit? Melo by to vyhodit chybu, ze jo? Je to podobne i v tech dalsich pripadech
```
  test("when hover on cat card it shows right title", async ({ page }) => {
    // await pages.hoverPage.catCard.hover();
    await expect(pages.hoverPage.catCardTitle).toHaveText("Kočka");
  });
```

Casto je fajn si zkontrolovat, ze opravdu testujes to, co se zmeni temi kroky - takze co se zmeni, ze najedes na danou karticku? 🤨 Je potreba na to vzdycky myslet.
A na to pomuze prave si zakomentovat dane kroky a bez tech by to melo failnout. :) 

Je to takhle jasne?

Reply:
Jo takže ten správný text už je daný těmi lokátory a pak dám jen:
test("when hover on cat card it shows right title", async ({ page }) => {
    await pages.hoverPage.catCard.hover();
    await expect(pages.hoverPage.catCardTitle).toBeVisible();

Reply:
Spíš on ten text tam je, ale není `visible`. Takže jsi testovala, jestli daná kartička má text "Kočka", což měla (ale pořád skrytý). Ale test byl o tom, že ten text měl být `visible`. 

Je to možná trošku matoucí, ale je potřeba si uvědomit, že v kódu ty věci jsou, ale nejsou vykreslené, takže jsou neviditelné. Ale v kódu to je a na to ses Ty ptala - má ten text?

Asi bychom na to mohli implementovat tohle meme
![image](https://github.com/user-attachments/assets/01ab0444-eb2b-448e-adc2-57635de090a6)

Z tohodle jen plyne, že někdy je dobré si otestovat, že moje assertace je vhodná - a to tím, že smažu krok(y) v testu a podívám se, jestli mi ten test spadne.

Reply:
Rozumím. Aktuálně to mám takto a když zakomentuju hover tak to padá a jinak to funguje. Tak myslim že teď už ty testy testují co mají.
test("when hover on cat card it shows right title", async ({ page }) => {
await pages.hoverPage.catCard.hover();
await expect(pages.hoverPage.catCardTitle).toBeVisible();

Reply:
jj, takhle je to super :) A pokud máš lokátor přes text, tak nemusíš řešit assert `toHaveText()`, protoze ti to spadne na lokatoru, ze to nemohlo najit. Pokud bys tam mela lokator pres ID nebo jinak, tak pak se hodi it `toHaveText()`